hphp/runtime/ext/hsl/ext_hsl_os.cpp

```diff
@@ -390,6 +390,17 @@ struct HSLFileDescriptor {
   }
 
   void close() {
+    for (auto obj : m_async_polls) {
+      auto wh = wait_handle<c_ExternalThreadEventWaitHandle>(obj.get());
+      if (!wh->isFinished()) {
+        auto file_await = static_cast<FileAwait*>(wh->getEvent());
+        file_await->unregisterHandler();
+        wh->cancel(create_object(s_ErrnoException, make_vec_array(
+          "Operation cancelled because FileDescriptor was closed.", ECONNABORTED)));
```

Comment:
- I don't think this should be `ECONNABORTED` - the stream_await/poll_async should largely be thought of as 'select, but async', and I think ECONNABORTED will be unexpected there.  Perhaps whatever's done here should be translated to ECONNABORTED by Hack code in the HSL though in accept_async
- I *think* EBADF is the most reasonable thing to happen here; AIUI it could happen anyway if a close hapened before the stream_await; otherwise EINTR seems like a good candidate